main/src/main/java/org/mobilitydata/gtfsvalidator/validator/NetworkIdConsistencyValidator.java

```diff
@@ -0,0 +1,175 @@
+package org.mobilitydata.gtfsvalidator.validator;
+
+import static org.mobilitydata.gtfsvalidator.notice.SeverityLevel.ERROR;
+
+import java.util.*;
+import javax.inject.Inject;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidationNotice;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.*;
+
+@GtfsValidator
+public class NetworkIdConsistencyValidator extends FileValidator {
+  private final GtfsRouteTableContainer routeTableContainer;
+  private final GtfsRouteNetworkTableContainer routeNetworkTableContainer;
+
+  private final GtfsNetworkTableContainer networkTableContainer;
+
+  @Inject
+  NetworkIdConsistencyValidator(
+      GtfsRouteTableContainer routes,
+      GtfsRouteNetworkTableContainer routeNetworks,
+      GtfsNetworkTableContainer networks) {
+    this.routeTableContainer = routes;
+    this.routeNetworkTableContainer = routeNetworks;
+    this.networkTableContainer = networks;
+  }
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    boolean hasNetworkIdField = this.routeTableContainer.hasColumn(GtfsRoute.NETWORK_ID_FIELD_NAME);
+
+    if (hasNetworkIdField && !this.routeNetworkTableContainer.isMissingFile()) {
+      noticeContainer.addValidationNotice(
+          new ConditionalForbiddenFileNotice(
+              GtfsRoute.FILENAME, GtfsRoute.NETWORK_ID_FIELD_NAME, GtfsRouteNetwork.FILENAME));
+    }
+    if (!this.networkTableContainer.isMissingFile() && hasNetworkIdField) {
+      noticeContainer.addValidationNotice(
+          new ConditionalForbiddenFileNotice(
+              GtfsRoute.FILENAME, GtfsRoute.NETWORK_ID_FIELD_NAME, GtfsNetwork.FILENAME));
+    }
+    validateUniqueRouteNetworkAssociation(noticeContainer);
+  }
+
+  private void validateUniqueRouteNetworkAssociation(NoticeContainer noticeContainer) {
+    Map<String, String> routeNetworkMap = buildRouteNetworkMap(noticeContainer);
+
+    for (GtfsRoute route : routeTableContainer.getEntities()) {
+      String routeId = route.routeId();
+      String networkId = route.networkId();
+      String existingRouteNetworkId = routeNetworkMap.get(routeId);
+
+      if (existingRouteNetworkId != null
+          && !existingRouteNetworkId.equals(networkId)
+          && networkId.length() > 0) {
+        int csvRowNumber =
+            routeNetworkTableContainer.byRouteIdMap().get(routeId).get(0).csvRowNumber();
+        noticeContainer.addValidationNotice(
+            new RouteNetworkAssociationDuplicateNotice(
```

Comment:
If it's already a validation error to specify both `routes.txt:network_id` and `route_networks.txt`, do we really need to check for different values between the two sources?

Reply:
@emmambd @tzujenchanmbd thoughts?

Reply:
after discussing with @emmambd and @tzujenchanmbd, we came to an agreement that the validation notice `route_networks_specified_in_more_than_one_file` combined with the fact that `route_id` is the primary key was good enough to cover the requirements of this story.